CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -216,6 +212,107 @@ function PlayerHospital:dailyAdviceChecks()
   end
 end
 
+--! Give advice about the player's progress towards winning or losing the level
+function PlayerHospital:giveLevelProgressAdvice()
+  -- Starting position
+  local level_config, level, start = self.world.map.level_config, self.world.map.level_number, {}
+  if level_config.towns and level_config.towns[level] then
+    start["balance"] = level_config.towns[level].StartCash
+    start["reputation"] = level_config.towns[level].StartRep
+  elseif level_config.town then
+    start["balance"] = level_config.town.StartCash
+    start["reputation"] = level_config.town.StartRep
+  end
+  if not start["balance"] then start["balance"] = 0 end
+  if not start["reputation"] then start["reputation"] = 500 end
+
+  -- Determine criteria of this level
+  self.world:determineWinningConditions()
+  local msgs, criteria_count = {}, 0
+  local win = {
+    reputation = self.world.goals[1]["win_value"],
+    balance = self.world.goals[2]["win_value"],
+    cured_p = self.world.goals[3]["win_value"], -- Patients cured percentage
+    cured_c = self.world.goals[4]["win_value"], -- Patients cured count
+    value = self.world.goals[6]["win_value"]
+  }
+  local lose = {
+    reputation = self.world.goals[1]["lose_value"],
+    balance = self.world.goals[2]["lose_value"],
+    killed_p = self.world.goals[5]["lose_value"] -- Patients killed percentage
+  }
```

Comment:
So a possible issue here is if the criterion is not set in the level file. Your checks will cause `attempt to compare nil with number`.

I have tricked myself a few times on trying to verify this so please clarify:
In base config the win/lose criteria is set but not against any value (all zeros).
```
  win_criteria = {
    [0] = {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
  },
  lose_criteria = {
    [0] = {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
    {Criteria = 0, MaxMin = 0, Value = 0, Group = 0, Bound = 0},
  },
```

However, none of these actually set values against e.g. criteria 6 (hospital value).
If we look at `EASY07.SAM` from Theme Hospital, only 4 win criteria are set (not pasting here).

As such, it looks like this PR doesn't accommodate for a not set value on these items.